beaglebone/thermistor/src/__main__.py

```diff
@@ -0,0 +1,167 @@
+        msgSend.thermistor = "white"
+    elif num == 1:
+        msgSend.thermistor = "blue"
+    elif num == 2:
+        msgSend.thermistor = "yellow"
+    global publishChannel
+    lcm.publish(publishChannel, msgSend.encode())
+
+
+def request_callback(channel, msg):
+    global outLock
+    with outLock:
+        whichTherm = ThermistorCmd.decode(msg)
+        global outputting
+        if whichTherm.thermistor == "white":
+            outputting = 0
+        elif whichTherm.thermistor == "blue":
+            outputting = 1
+        elif whichTherm.thermistor == "yellow":
+            outputting = 2
+
+
+def readVolt():
+    V2 = []
+    for i in range(3):
+        # Documentation says need to read twice due to bug in the library
+        temp = adc.read(adcPin[i])
+        temp = adc.read(adcPin[i])
+        # Adjusting V2 to actual Voltage (Multiply by 1.8V since that is max ADC can handle)
+        temp *= 1.8
+        V2.append(temp)
+    return V2
+
+
+def calcRTherm(V2):
+    Rt = []
+    for i in range(3):
+        Rt.append(((R1[i]*V1)/(V2[i]))-(R1[i]))
+    return Rt
+
+
+def readTherms():
+    # reading voltage from ADC
+    V2 = readVolt()
+
+    # calculating R2 (resistance of thermistor)
+    Rt = calcRTherm(V2)
+
+    currTemp = []
+
+    for i in range(3):
+        # Determine which set of constants should be used
+        if(Rt[i] < 692600 and Rt[i] >= 32770):
+            constantSet = 0
+        elif(Rt[i] < 32770 and Rt[i] >= 3599):
+            constantSet = 1
+        elif(Rt[i] < 3599 and Rt[i] >= 681.6):
+            constantSet = 2
+        elif(Rt[i] < 681.6 and Rt[i] >= 187):
+            constantSet = 3
+        else:
+            # TODO proper error handling (out of bounds temp)
+            print("OOB temp")
+
+        # Using the Rt/R25 range to determine actual temp
+        lnRtoverR25 = ln(Rt[i]/R25)
+        oneOverT = constantArray[constantSet][0] + (constantArray[constantSet][1] * lnRtoverR25) \
+            + (constantArray[constantSet][2] * lnRtoverR25 * lnRtoverR25) \
+            + (constantArray[constantSet][3] * lnRtoverR25 * lnRtoverR25 * lnRtoverR25)
+
+        # calc actual temp from the IoverT value the formula gives
+        currTemp.append((1 / oneOverT) - 272.15)  # -272.15 for K to C
+
+    return currTemp
+
+
+# All this function does is LCM handle on its own thread
+def handleLCM():
+    while True:
+        lcm.handle()
+
+
+def runTherms():
+    # main while loop
+    while True:
+        currTemp = readTherms()
+
+        # publishing message and waiting desired interval
+        global outputting
+        if outputting != -1:
+            publishMessage(currTemp[outputting], outputting)
+        sleep(LCM_INTERVAL)
+
+
+def main():
+    # Start the LCM
+    global lcm
+    lcm = lcm_.LCM()
+
+    # init the ADC
+    adc.setup()
+
+    lcm.subscribe("/thermistor_cmd", request_callback)
+
+    # Setting up lock for outputting
+    global outLock
+    outLock = threading.Lock()
```

Comment:
What's the point of this lock?

outlock.lock() would force other threads that call outlock.lock() to wait, but I don't see any other functions using it